gnomad_qc/v4/subset.py

```diff
@@ -1,269 +1,471 @@
+            subset_workspaces=args.subset_workspaces,
+            tmp_dir=args.tmp_dir,
+        )
 
-    return selected_variant_qc_annotations
 
+def get_gnomad_datasets(data_type: str, n_partitions: Optional[int], test: bool):
+    """
+    Get requested data type's v4 VariantDataset.
 
-def main(args):
-    """Filter the gnomAD v4 VariantDataset to a subset of specified samples."""
-    hl.init(
-        log="/v4_subset.log",
-        default_reference="GRCh38",
-        tmp_dir="gs://gnomad-tmp-4day",
-    )
-    test = args.test
-    output_path = args.output_path
-    header_dict = HEADER_DICT
-    add_variant_qc = args.add_variant_qc
-    variant_qc_annotations = args.variant_qc_annotations
-    pass_only = args.pass_only
-    split_multi = args.split_multi
-
-    if args.vcf and not split_multi:
-        raise ValueError(
-            "VCF export without split multi is not supported at this time."
+    :param data_type: Type of data to subset.
+    :param n_partitions: Number of desired partitions for the VDS, repartioned on read.
+    :param test: Whether to filter to the first 2 partitions for testing.
+    :return: The gnomAD v4 VariantDataset.
+    """
+    if data_type == "exomes":
+        vds = get_gnomad_v4_vds(
+            n_partitions=n_partitions, remove_hard_filtered_samples=False
         )
-    if (add_variant_qc or pass_only) and not split_multi:
-        raise ValueError(
-            "Cannot annotate variant QC annotations or filter to PASS variants on an"
-            " unsplit dataset."
+    else:
+        vds = get_gnomad_v4_genomes_vds(
+            n_partitions=n_partitions, remove_hard_filtered_samples=False
         )
 
-    vds = get_gnomad_v4_vds(
-        n_partitions=args.n_partitions, remove_hard_filtered_samples=False
-    )
-
     if test:
         vds = hl.vds.VariantDataset(
             vds.reference_data._filter_partitions(range(2)),
             vds.variant_data._filter_partitions(range(2)),
         )
+    meta_ht = meta(data_type=data_type).ht()
+
+    return vds, meta_ht
+
+
+def get_subset_ht(
+    subset_samples: Optional[str],
+    subset_workspaces: Optional[str],
+    meta_ht: Optional[hl.Table],
+) -> hl.Table:
+    """
+    Get the subset HT.
 
-    meta_ht = meta().ht()
-    subset_ht = hl.import_table(args.subset_samples or args.subset_workspaces)
+    :param subset_samples: Path to a text file with sample IDs for subsetting and a header: 's'.
+    :param subset_workspaces: Path to a text file with Terra workspaces that should be included in the subset and a header: 'terra_workspace'.
+    :param meta_ht: The meta HT.
+    :param vds_cols: The VDS columns.
+    :return: The subset HT.
+    """
+    if subset_workspaces and not meta_ht:
+        raise ValueError("A meta HT must be provided if subsetting by workspaces.")
 
-    if args.subset_workspaces:
+    if subset_workspaces:
         terra_workspaces = hl.literal(subset_ht.terra_workspace.lower().collect())
```

Comment:
```suggestion
        terra_workspaces = hl.literal(meta_ht.terra_workspace.lower().collect())
```
I completely missed this earlier

Reply:
Oh its suppose to be subset_workspaces